src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -0,0 +1,25 @@
+import argparse
+from pathlib import Path
+
+import pytest
+
+from diffpy.labpdfproc.tools import set_output_directory
+
+params1 = [
+    ([None], [Path.cwd()]),
+    (["corrected"], [Path("corrected").resolve()]),
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params1)
+def test_set_output_directory(inputs, expected):
+    expected_output_directory = expected[0]
+    actual_args = argparse.Namespace(output_directory=inputs[0])
+    actual_args.output_directory = set_output_directory(actual_args)
+    assert actual_args.output_directory == expected_output_directory
+    assert Path(actual_args.output_directory).exists()
+    assert Path(actual_args.output_directory).is_dir()
+
+    with pytest.raises(FileExistsError):
```

Comment:
I think the code is easier to read if you make a new function called `test_set_output_directory_bad()` and then test the cases that should fail.  Then to make it easy to read, create the file in the test and then test that it the exception is raised by the function.

I wonder though if we want to catch any case where `Path().exists()` returns true and `Path().is_dir()` returns false?  Isn't that better?

Reply:
I added the tests. I think the case where Path().exists() is true and Path().is_dir() is false refers to the situation when user provides an existing file directory, so I added it in the bad case. On the other hand, if user writes the output directory as new_dir.py, I count it as one of the acceptable cases (the function creates a directory called new_dir.py). Please let me know if this is not a good idea.

Reply:
That is certainly one situation it covers, but I think anything that causes a crash is already covered.  We only need to handle the case where we are handling an Exception (not letting it crash) or some other "magic" situation where the code could behave unexpectedly, for example, the users requests an output directory but the code doesn't create it for some reason that doesn't cause a crash and then it writes in the current directory, or sthg.

So testing for "Path().exists() returns true and Path().is_dir()" and raising something like a RunTimeError with a helpful error message if this fails.  To test it we would have to  come up with a way of accessing that situation.  I can't think of any, but having that in the code makes me feel more secure because just because I can't think how to do it doesn't mean it can't be done.....